### PR TITLE
Add delete all azure

### LINKED ISSUE #1377: Add a delete all option for the azure_vm
Add an delete option to delete all related resources of a vm, user could decide whether delete all resources attached to a single vm

### PR DESCRIPTION
**Pull request by JunqiZhang0**
Fix #1377

### CODE REVIEW

File: linchpin/provision/roles/azure/tasks/provision_azure_vm.yml [hunk 1]
```diff
@@ -70,7 +70,7 @@
     tenant: "{{ auth_var['tenant'] | default(omit) }}"
     secret: "{{ auth_var['secret'] | default(omit) }}"
     subscription_id: "{{ auth_var['subscription_id'] | default(omit) }}"
-    admin_username: "{{ res_def['vm_username'] | default('{{linchpinUsername}}') }}"
+    admin_username: "{{ res_def['vm_username'] | default(linchpinUsername) }}"
```
Comment (samvarankashyap) [inline linchpin/provision/roles/azure/tasks/provision_azure_vm.yml]: @junqizhang Where are linchpinUsername and linchpinPassword coming from ???

Comment (samvarankashyap) [reply linchpin/provision/roles/azure/tasks/provision_azure_vm.yml]: Shouldnt they be in single quotes ??

Comment (JunqiZhang0) [reply linchpin/provision/roles/azure/tasks/provision_azure_vm.yml]: They r coming from the defaults.yml

**Comment by JunqiZhang0** [comment]
[merge]

**Pull request merged**